rust/agama-lib/src/utils/transfer/handlers/yast.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) [2025] SUSE LLC
+//
+// All Rights Reserved.
+//
+// This program is free software; you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 2 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, contact SUSE LLC.
+//
+// To contact SUSE LLC about this file by physical or electronic mail, you may
+// find current contact information at www.suse.com.
+
+use std::io::Write;
+
+use url::Url;
+
+use crate::utils::{
+    transfer::{file_finder::FileFinder, file_systems::FileSystemsList},
+    TransferError, TransferResult,
+};
+
+/// Handler for the cd:, dvd: and hd: schemes
+///
+/// It converts those schemes to a regular DeviceHandler.
+#[derive(Default)]
+pub struct HdHandler {}
+
+impl HdHandler {
+    pub fn get(&self, url: Url, out_fd: &mut impl Write) -> TransferResult<()> {
+        let device = url.query_pairs().find(|(key, _value)| key == "devices");
+
+        let Some((_, device_name)) = device else {
+            return Err(TransferError::MissingDevice(url));
+        };
+        let device_name = device_name.strip_prefix("/dev/").unwrap();
```

Comment:
what about allowing to just pass hd name instead of full "/dev/hda1" ? Code wise it will be easy:
```diff
        let device_name = device_name.strip_prefix("/dev/").unwrap_or(device_name);
```